drivers/rail/rail_netdev.c

```diff
@@ -0,0 +1,856 @@
+               change the attribute in the netdev struct */
+            assert(len == sizeof(uint16_t));
+            uint8_t chan = (((const uint16_t *)val)[0]) & UINT8_MAX;
+
+            if (dev->params.freq == RAIL_TRANSCEIVER_FREQUENCY_2P4GHZ) {
+                if (chan < RAIL_2P4GH_MIN_CHANNEL || chan > RAIL_2P4GH_MAX_CHANNEL) {
+                    res = -EINVAL;
+                    break;
+                }
+            }
+            else if (dev->params.freq == RAIL_TRANSCEIVER_FREQUENCY_868MHZ) {
+                /* 868MHz has only one channel, channel 0! */
+                if (chan != RAIL_868MHZ_DEFAULT_CHANNEL) {
+                    res = -EINVAL;
+                    break;
+                }
+            }
+            else if (dev->params.freq == RAIL_TRANSCEIVER_FREQUENCY_912MHZ) {
+                if (chan > RAIL_912MHZ_MAX_CHANNEL) {
+                    res = -EINVAL;
+                    break;
+                }
+            }
+            else {
+                res = -EINVAL;
+                LOG_ERROR("Unknown radio frequency configured\n");
+                assert(false);
+                break;
+            }
+            /* since we have to provide the channel for each tx or rx, just
+               change the attribute in the netdev struct */
+            /* TODO if chan_old != chan_new -> interupt rx op? */
+            /* don't set res to set netdev_ieee802154_t::chan */
+            dev->netdev.chan = chan;
+
+            break;
+        case (NETOPT_CHANNEL_PAGE):
+            /* TODO?? */
+            break;
+        case (NETOPT_ADDRESS):
+            assert(len <= sizeof(uint16_t));
+            /* RIOT uses Big endian, RAIL driver blob little endian ... */
+
+            le_u16 = ntohs(*((const uint16_t *)val));
+
+            /* RAIL driver blob can manage upto RAIL_IEEE802154_MAX_ADDRESSES
+               TODO how does RIOT handle multible short addresses?
+               atm just at pos 0
+             */
+            rail_ret = RAIL_IEEE802154_SetShortAddress(dev->rhandle, le_u16, 0);
+
+            if (rail_ret != RAIL_STATUS_NO_ERROR) {
+                LOG_ERROR("[rail] error setting short address: msg: %s\n",
+                          rail_error2str(rail_ret));
+                res = -EFAULT;
+                break;
+            }
+
+            /* don't set res to set netdev_ieee802154_t::short_addr */
+            break;
+        case (NETOPT_ADDRESS_LONG):
+            assert(len <= sizeof(uint64_t));
+            /* RAIL driver blob can manage upto RAIL_IEEE802154_MAX_ADDRESSES
+               TODO how does RIOT handle multible long addresses?
+               atm just at pos 0
+             */
+            /* now the long addr ... the RAIL API docu says it have to be in
+               "over the air byte order", therefore little endian again ... */
+            le_u64 = byteorder_swapll(*((const uint64_t *)val));
+
+
```

Comment:
Extra new line

Reply:
OK. I removed the double newlines in all files.